### DISCUSSION TITLE
TemplatedEmail - inlined image shows up as attachment

### DISCUSSION THREAD

**Discussion by DavidPetrasek** [symfony/symfony#50409] [Q&A] [answered]
I can't find a way to inline an image inside an email message and **not** show it as an attachment.
I tried both ways:

1. Twig function:
` <img src="{{ email.image('@imgDir/logo/logo.svg') }}" alt="Logo">`

2. Manually setting the CID: 
```
$email = (new TemplatedEmail())
        ->addPart((new DataPart(new File($this->projectDir.'/assets/img/logo/logo.svg'), 'logo', 'image/svg+xml'))->asInline());
```
Then in twig: `<img src="cid:logo" alt="Logo">`

**Comment by henderkes** [comment]
Sounds like your email client has a problem displaying the mail, since solution number 1 should work (and does work fine for me). You should set the width and height attributes, though.

**Reply by DavidPetrasek** [reply]
Tried setting the width and height attribute and it changed nothing. 
E-mail clients used: https://outlook.live.com, MailCatcher.

When I used the library [PHPMailer](https://github.com/PHPMailer/PHPMailer) in other project, It worked everywhere as expected - the image was not shown as an attachment. The code I used there:
`$phpMailer->addEmbeddedImage ('/absolute/path/to/image', 'logo1');`
Then in the template **I didn't have to set the width nor height attribute**:
`<img src="cid:logo1" alt="logo" title="logo"/>`

**Reply by henderkes** [reply]
What's the html code that rendering the mail generates?

**Reply by DavidPetrasek** [reply]
Here is the source code of the received message:
[source.txt](https://github.com/symfony/symfony/files/11685258/source.txt)
One thing is different now - recently I had to change the path `@imgDir/logo/logo.svg` to `@imgDir/logo/anonymne.svg`

**Reply by henderkes** [reply]
Not sure how the weird =s or line breaks happened, but other than those this is perfectly valid html to show the image inline and not as an attachment.

**Reply by DavidPetrasek** [reply]
[Here I uploaded](https://drive.google.com/file/d/1eJVW2i6Cw-kaZB0YFScmoLBEwkmZbwqJ/view?usp=sharing) the EML version. I opened it in Mozilla Thunderbird and here the attachment is now **correctly not being shown**. So, apparently only web-based email clients like Outlook and Gmail are not able to hide the attachment.

Then I compared the EML message ([download](https://drive.google.com/file/d/1gTczzKMYSUwLwF338fYEPdt42WmZiAKC/view?usp=sharing)) received from PHPMailer, which works even in web-based email clients. I found that message sent by PHPMailer has `Content-Transfer-Encoding` always set to `base64`, whilst Symfony Mailer sets only the inlined image to `base64` and the rest is being set to `quoted-printable`.

Does anybody know how to change `Content-Transfer-Encoding` in Symfony Mailer?
I tried this, but it had no effect:
```
$email = (new TemplatedEmail())
       ->to(.....
 $email->getHeaders()
            ->addTextHeader('Content-Transfer-Encoding', 'base64');
```
It seems like it is not possible now: [#16812](https://github.com/symfony/symfony-docs/issues/16812).

**Reply by DavidPetrasek** [reply, marked as answer]
**SOLUTION:**
Problem was the SVG format of the image, which is not supported everywhere. It works now with PNG.

**Comment by MatTheCat** [comment]
What I’m sure of is that an embedded file always is an attachment. Then the client will decide whether it shows such attachments. Unfortunately I have no idea how they’re doing it :sweat_smile: 

Friendly ping @hteumeuleu!

**Comment by pavel-macovschi** [comment]
Hello. How do you send TemplatedEmail sync or async  ? 

Do you use BodyRendererInterface before sending and signing like DKIMSigner  ? 

I ask you because maybe I have similar issue, take a look at:  [#50685](https://github.com/symfony/symfony/issues/50685)

**Reply by DavidPetrasek** [reply]
I send all e-mails as **synchronous**.
I don't use BodyRendererInterface nor DkimSigner anywhere in the project.
I will take a look at your issue.
